examples/Mediawiki/PageProtection.yaml

```diff
@@ -0,0 +1,170 @@
+      form-data:
+        action: protect
+        title: Protected Page
+        token: p/var: token
+        protections: edit=sysop
+        format: json
+    response:
+      body:
+        protect:
+          title: Protected Page
+          protections:
+            edit: sysop
+            expiry: infinite
+  # Add edit protections to only allow auto confirmed users to edit Semi Protected Page
+  - use-session: RootUser
+    request:
+      method: post
+      path: api.php
+      form-data:
+        action: protect
+        title: Semi Protected Page
+        token: p/var: token
+        protections: edit=autoconfirmed
+        format: json
+    response:
+      body:
+        protect:
+          title: Semi Protected Page
+          protections:
+            edit: autoconfirmed
+            expiry: infinite
+
+tests:
+  - description: Root user should be able to edit Protected Page
+    interaction:
+      - inherit: edit
+        use-session: RootUser
+        request:
+          parameters:
+            title: Protected Page
+          form-data:
+            summary: test edit
+            text: Protected Page edited by RootUser
+            token: p/var: token
+  - description: Root user should be able to edit Semi Protected Page
+    interaction:
+      - inherit: edit
+        use-session: RootUser
+        request:
+          parameters:
+            title: Semi Protected Page
+          form-data:
+            summary: test edit
+            text: Semi Protected Page edited by RootUser
+            token: p/var: token
+  - description: Autoconfirmed user should not be able to edit Protected Page
+    interaction:
+      - inherit: edit
+        use-session: Alice
+        request:
+            parameters:
+              title: Protected Page
+            form-data:
+              summary: test edit
+              text: Protected Page edited by Alice
+              token: p/var: Alice/edit-token
+          response:
+            body:
+              code: protectedpage
+              info: This page has been protected to prevent editing or other actions.
+  - description: Autoconfirmed user should be able to edit Semi Protected Page
+    interaction:
+      - inherit: edit
+        use-session: Alice
+        request:
+          parameters:
+            title: Semi Protected Page
+          form-data:
+            summary: test edit
+            text: Semi Protected Page edited by Alice
+            token: p/var: Alice/edit-token
+  - description: Anonymous user should not be able to edit Protected Page
+    interaction:
+      - inherit: edit
+        request:
+            parameters:
+              title: Protected Page
+            form-data:
+              summary: test edit
+              text: Protected Page edited by Anonymous
+              token: +\
+          response:
+            body:
+              code: protectedpage
+              info: This page has been protected to prevent editing or other actions.
```

Comment:
Checking the info isn't too relevant, and it's brittle to changes of the localization message. The exact message is not part of the contract, so it should not be part of the test.

If you really need to check something about a message, turn off localization by setting uselang=qqx. That causes messages to be generated unlocalized, as the key and parameters. Try it on wikipedia: https://en.wikipedia.org/wiki/Main_Page?uselang=qqx

Not needed here, because the code is sufficient.

Reply:
Done.